src/demo-app/stepper/stepper-demo.html

```diff
@@ -1,3 +1,52 @@
+<h2>Linear Vertical Stepper Demo</h2>
+<form [formGroup]="formGroup" novalidate>
+  <div formArrayName="formArray">
+    <md-vertical-stepper>
+      <md-step>
+        <div [formGroupName]="0">
+          <ng-template mdStepLabel>Fill out your name</ng-template>
+          <md-input-container>
+            <input mdInput placeholder="First Name" formControlName="firstNameFormCtrl" required>
+            <md-error>This field is required</md-error>
+          </md-input-container>
+
+          <md-input-container>
+            <input mdInput placeholder="Last Name" formControlName="lastNameFormCtrl" required>
+            <md-error>This field is required</md-error>
+          </md-input-container>
+          <div>
+            <button md-button mdStepperNext>Next</button>
+          </div>
+        </div>
+      </md-step>
+
+      <md-step [disabled]="!formGroup.controls.formArray.controls[0].valid">
+        <div [formGroupName]="1">
+          <ng-template mdStepLabel>
+            <div>Fill out your phone number</div>
+          </ng-template>
+          <md-input-container>
+            <input mdInput placeholder="Phone number" formControlName="phoneFormCtrl" required>
+            <md-error>This field is required</md-error>
+          </md-input-container>
+          <div>
+            <button md-button mdStepperPrevious>Back</button>
+            <button md-button mdStepperNext>Next</button>
+          </div>
+        </div>
+      </md-step>
+
+      <md-step [disabled]="!formGroup.controls.formArray.controls[1].valid">
```

Comment:
What about creating a custom validator function for the group or array? It seems like you need to consider more than one form control here, which is the classic situation where that is needed. A higher-level validator would allow you to surface more information in a custom error message, like all step validity or the latest step where all steps before it are valid.